documentcloud.php

```diff
@@ -3,379 +3,476 @@
-                )) 
-            ) { return; }
-        
-            $default_sizes = $this->get_default_sizes();
-            $default_atts = $this->get_default_atts();
-            $wide_assets = get_post_meta($post_id, 'wide_assets', true);
-            $matches = array();
-                
-            preg_match_all('/'.get_shortcode_regex().'/', $post->post_content, $matches);
-            $tags = $matches[2];
-            $args = $matches[3];
-            foreach($tags as $i => $tag) {
-                if ($tag == "documentcloud") {
-                    $parsed_atts = shortcode_parse_atts($args[$i]);
-                    $atts = shortcode_atts($default_atts, $parsed_atts);
-
-                    // get a doc id to keep array keys consistent
-                    if (isset($atts['url'])) {
-                        $elements = $this->parse_dc_url($atts['url']);
-                        if (isset($elements['document_slug'])) {
-                            $meta_key = $elements['document_slug'];
-                            if (isset($elements['page_number'])) {
-                                $meta_key .= "-p{$elements['page_number']}";
-                            }
-                            else if (isset($elements['note_id'])) {
-                                $meta_key .= "-a{$elements['note_id']}";
-                            }
-                        }
-                    } else if (isset($atts['id'])) {
-                        $meta_key = $atts['id'];
-                    }
-                
-                    // if no id, don't bother storing because it's wrong
-                    if (isset($meta_key)) {
-                        $width = intval(isset($parsed_atts['width']) ? $parsed_atts['width'] : $atts['maxwidth']);
-                        if ($atts['format'] == "wide" || $width > $default_sizes['width']) {
-                            $wide_assets[$meta_key] = true;
-                        } else {
-                            $wide_assets[$meta_key] = false;
-                        }
-                    }
-                }
-            }
-            update_post_meta($post_id, 'wide_assets', $wide_assets);
-        }
-    }
-    
+	// Plugin constants
+	const CACHING_ENABLED		   = true,
+		  DEFAULT_EMBED_FULL_WIDTH = 940,
+		  OEMBED_RESOURCE_DOMAIN   = 'www.documentcloud.org',
+		  OEMBED_PROVIDER		   = 'https://www.documentcloud.org/api/oembed.{format}',
+		  DOCUMENT_PATTERN		   = '^(?P<protocol>https?):\/\/www\.documentcloud\.org\/documents\/(?P<document_slug>[0-9]+-[a-z0-9-]+)';
+	/**
+	 * Constructor.
+	 */
+	function __construct() {
+		// Check for conflicts with other DocumentCloud plugins.
+		// Not needed on WordPress VIP since no other DocumentCloud plugins exist.
+		if ( ! defined( 'WPCOM_IS_VIP_ENV' ) || ! WPCOM_IS_VIP_ENV ) {
+			add_action( 'admin_init', array( $this, 'check_dc_plugin_conflict') );
+			add_action( 'init', array( $this, 'register_dc_oembed_provider' ) );
```

Comment:
@bcampeau We'll still need to register DC as an oEmbed provider, so this line needs to be outside the conditional block. Unless VIP also means we're whitelisted as an oEmbed provider, which would be *wonderful*.